core/src/main/java/org/apache/gravitino/catalog/CatalogManager.java

```diff
@@ -474,6 +488,65 @@ public void testConnection(
     }
   }
 
+  @Override
+  public void activateCatalog(NameIdentifier ident)
+      throws NoSuchCatalogException, NonInUseEntityException {
+    try {
+      boolean inUse = catalogInUse(store, ident);
+      if (!inUse) {
+        store.update(
+            ident,
+            CatalogEntity.class,
+            EntityType.CATALOG,
+            catalog -> {
+              CatalogEntity.Builder newCatalogBuilder =
+                  newCatalogBuilder(ident.namespace(), catalog);
+
+              Map<String, String> newProps =
+                  catalog.getProperties() == null
+                      ? new HashMap<>()
+                      : new HashMap<>(catalog.getProperties());
+              newProps.put(PROPERTY_IN_USE, "true");
+              newCatalogBuilder.withProperties(newProps);
+
+              return newCatalogBuilder.build();
+            });
+      }
```

Comment:
```suggestion
if (inUse) {
   return;
}

// Other code

```

You can use this to reduce nesting.